app/src/main/java/com/madonasyombua/growwithgoogleteamproject/ui/fragment/LoginFragment.java

```diff
@@ -35,12 +45,19 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
         binding.loginFragmentButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                String username_text = extractText(binding.editUsername);
+                String email_text = extractText(binding.editEmail);
                 String password_text = extractText(binding.editPassword);
 
-                Toast.makeText(getActivity(), "Logging in", Toast.LENGTH_SHORT).show();
-                //TODO: start MainActivity from Here
-                startActivity(new Intent(getActivity(), MainActivity.class));
+                FirebaseUser currentUser = LoginManager.signinUser(getActivity(), new User(email_text, password_text));
+
+                if(currentUser!=null) {
+
+
+                    Toast.makeText(getActivity(), "Logging in", LENGTH_SHORT).show();
```

Comment:
Toast should not display progress. Either "Logged In" or "Error Logging In". Use a ProgressBar to indicate you are trying to log in.

Reply:
Good job team mates :) 💯

Reply:
@MaherSoua You could do a button that also shows progress like this:
https://github.com/leandroBorgesFerreira/LoadingButtonAndroid

Or just show a dialog with status and progressbar in it. I think those would be best UX wise

Reply:
Nice idea, but I prefer do it in another PR :)

Reply:
That looks lovely @adevezin , @MaherSoua awesome.